### PR TITLE
fix duplicate watch channel task

### PR DESCRIPTION
**Pull request by weiliu1031**
Signed-off-by: Wei Liu <wei.liu@zilliz.com>
Related to https://github.com/milvus-io/milvus/issues/20580

/kind bug

### CODE REVIEW

File: internal/querycoordv2/task/task_test.go [hunk 1]
```diff
@@ -158,6 +160,48 @@ func (suite *TaskSuite) BeforeTest(suiteName, testName string) {
...
+			CollectionID: suite.collection,
+			Channel:      channel,
+		})
+	}
+	suite.dist.LeaderViewManager.Update(targetNode, views...)
+
+	for _, task := range tasks {
+		err := suite.scheduler.Add(task)
+		suite.Error(err)
+		suite.ErrorIs(err, ErrTaskAlreadyDone)
```
Comment (yah01) [inline internal/querycoordv2/task/task_test.go:201]: Only keeping this line is fine, 200 is redundant

File: internal/querycoordv2/task/task_test.go [hunk 2]
```diff
@@ -399,6 +443,47 @@ func (suite *TaskSuite) TestLoadSegmentTask() {
...
+	}
+	for _, segment := range suite.loadSegments {
+		view.Segments[segment] = &querypb.SegmentDist{NodeID: targetNode, Version: 0}
+	}
+	suite.dist.LeaderViewManager.Update(targetNode, view)
+
+	for _, task := range tasks {
+		err := suite.scheduler.Add(task)
+		suite.Error(err)
+		suite.ErrorIs(err, ErrTaskAlreadyDone)
```
Comment (yah01) [inline internal/querycoordv2/task/task_test.go:484]: Same as line 201

**Comment by weiliu1031** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/approve

**Pull request merged**